more_itertools/recipes.py

```diff
@@ -355,6 +355,27 @@ def unique_everseen(iterable, key=None):
     Sequences with a mix of hashable and unhashable items can be used.
     The function will be slower (i.e., `O(n^2)`) for unhashable items.
 
+    Remember that ``list`` objects are unhashable - you may want to convert
+    an iterable of ``list`` objects to an iterable of ``tuple`` objects
+    to avoid slowdown. S
+
+        >>> slow_iterable = [1, 2], [2, 3], [1, 2]
+        >>> list(unique_everseen(slow_iterable))
+        [[1, 2], [2, 3]]
+        >>> fast_iterable = (tuple(x) for x in slow_iterable)
+        >>> list(unique_everseen(fast_iterable))
+        [(1, 2), (2, 3)]
+
+    Similary, you may want to convert unhashable ``dict`` objects using
+    the :func:`dict.items` method.
+
+        >>> slow_iterable = {'a': 'b'}, {'c': 'd', 'e': 'f'}, {'a': 'b'}
+        >>> list(unique_everseen(slow_iterable))
+        [{'a': 'b'}, {'c': 'd', 'e': 'f'}]
+        >>> fast_iterable = (tuple(sorted(x.items())) for x in slow_iterable)
```

Comment:
Note that `sorted` really relies on the values to be comparable which may not be the case. In some cases it might even be the case that the objects don't define a total ordering which may lead to unexpected results.